Multiple Maximum Generation and Pumping for Hydro (Refactoring)

nikolaredstork opened the pull request:
This PR is created for clearing comments from RTE on feature/max-hydro-gen-pump-per-h-cr23 branch.
Link of the base PR :
https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/1470

nikolaredstork committed:
Power Credits comments and strings changed to Max Power

nikolaredstork committed:
Changing name convention

nikolaredstork committed:
Implicit conversion code smell

nikolaredstork committed:
Files name modification

nikolaredstork committed:
Renaming matrices from PartHydro and DataTransfer classes

nikolaredstork committed:
Renaming matrices from DataSeriesHydro class

nikolaredstork committed:
HydroMaxTSReader class

nikolaredstork committed:
Refactoring

nikolaredstork committed:
New LoadMaxPower function

nikolaredstork committed:
bugfix

nikolaredstork committed:
Excluding unnecessary data from tests and bugfix

nikolaredstork committed:
Code smell

flomnes commented on src/solver/hydro/management/management.h:
```diff
@@ -88,7 +88,7 @@ struct TmpDataByArea
 {
     std::vector<double> HydrauliqueModulableQuotidien; /* indice par jour */
-    std::vector<double> NiveauxReservoirsDebutJours;   //Niveaux (quotidiens) du reservoir de d�but
+    std::vector<double> NiveauxReservoirsDebutJours;   //Niveaux (quotidiens) du reservoir de d�but
```
```suggestion
    std::vector<double> NiveauxReservoirsDebutJours;   //Niveaux (quotidiens) du reservoir de début
```

nikolaredstork replied:
Done!

flomnes commented on src/libs/antares/study/parts/hydro/hydromaxtimeseriesreader.h:
```diff
@@ -54,17 +58,19 @@ class DataTransfer
-    bool SupportForOldStudies(Study& study, const AnyString& folder, Area& area);
-
-
+private:
+    bool LoadDailyMaxPowersAndEnergies(const AnyString& folder, const Area& area);
+    bool SaveDailyMaxEnergy(const AnyString& folder, Area& area);
+    bool SaveDailyMaxPowerAsHourly(const AnyString& folder, Area& area);
+    bool SaveMaxGenerationEnergy(const AnyString& folder, Area& area);
+    bool SaveMaxPumpingEnergy(const AnyString& folder, Area& area);
+    bool SaveDailyMaxGenPowerAsHourly(const AnyString& folder, Area& area);
+    bool SaveDailyMaxPumpPowerAsHourly(const AnyString& folder, Area& area);
```
Add a comment about legacy ? For versions newer than XXX, use YYY
For older versions, use ZZZ

nikolaredstork replied:
All of these functions are used when loading old study! HydroMaxTimeSeriesReader class is just for supporting old studies.

flomnes commented on src/libs/antares/study/parts/hydro/hydromaxtimeseriesreader.cpp:
```diff
@@ -0,0 +1,214 @@
+
+    while (hours < HOURS_PER_YEAR && days < DAYS_PER_YEAR)
+    {
+        for (uint i = 0; i < 24; ++i)
```
```suggestion
        for (uint i = 0; i < HOURS_PER_DAY; ++i)
```
?

nikolaredstork replied:
I've implemented that macro earlier today when i was refactoring AreaScratchpad class, but I did not pushed it. Regarding AreaScratchpad class will you be able to give me your comment on this?
https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/1470#discussion_r1333034108

nikolaredstork committed:
Daily mean maximum generation/pumping matrices

nikolaredstork committed:
Check Power Bounds

nikolaredstork committed:
Rewrite comment

nikolaredstork committed:
Name convention

nikolaredstork committed:
Code smells

nikolaredstork committed:
Comments and function name changed

nikolaredstork committed:
Lambda function refactor

nikolaredstork committed:
Get and Set functions

nikolaredstork committed:
Small refactoring

nikolaredstork committed:
Name convention

nikolaredstork committed:
Ternary operator check index excluded

nikolaredstork committed:
Clang Format undone

nikolaredstork committed:
Function argument names

nikolaredstork committed:
Possible Bug?

nikolaredstork committed:
Refactoring lambda function

guilpier-code commented on src/libs/antares/study/parts/hydro/series.h:
```diff
@@ -153,25 +154,103 @@ class DataSeriesHydro
+        void resizeWhenOneTS(Area& area, uint nbTimeSeriesSup_);
+
+    private:
+        Matrix<double, int32_t>& maxHourlyGenPower;
```
Please keep the trailing underscores for private variables, not for methods arguments.

nikolaredstork replied:
Done!

nikolaredstork committed:
Refactoring Unit tests

nikolaredstork committed:
Refactoring unit tests

nikolaredstork committed:
Unit tests

flomnes commented on src/tests/src/libs/antares/study/parts/hydro/CMakeLists.txt:
```diff
@@ -1,76 +1,66 @@
 # Useful variables definitions
 set(src_libs_antares_study "${CMAKE_SOURCE_DIR}/libs/antares/study")
 set(src_libs_antares "${CMAKE_SOURCE_DIR}/libs/antares")
+set(src_test_utilis "${CMAKE_SOURCE_DIR}/tests/src/utlis")
```
I think there are two typos
```suggestion
set(src_test_utils "${CMAKE_SOURCE_DIR}/tests/src/utils")
```

nikolaredstork replied:
Done!

nikolaredstork committed:
Typo and renaming

nikolaredstork committed:
Unit test fix and renaming

nikolaredstork committed:
hydro reader unit tests refactoring

nikolaredstork committed:
Refactoring unit tests help functions

nikolaredstork committed:
Excluding saving new files before upgrade

nikolaredstork committed:
Renameing files

nikolaredstork committed:
Get function instead of class data member

guilpier-code committed:
Refactor hydro max power TS consistency (#1707)

* Refactor max power TS checks : several things (see commit message)

This commit is large, it should be shorter, but the long compilation time forces to gather the devs in one piece.
- Remove the exception raising, not necessary : the gotFatalError (set to true) will trigger an exception anyway.
- class NbTsComparer : is generalized to any time series pair :
   + to isolate it from business context
   + to have a chance to re-use it in another context
- Move both classes NbTsComparer and TsActions outside class DataSeriesHydro, for same reasons as above
- Rename TsActions::handleBothGreaterThanOne (unclear) into something more meaningful
- Consequences of previous changes in tests

* Refactor max power TS checks : several things (see commit message)

TsActions::areaToInvalidate :
- Remove areaToInvalidate from class TsActions (it has nothing to do in here) and make it a free function
- Extract areaToInvalidate calls from TsActions::resizeWhenOneTS, as calling it here not related to TsActions

* Refactor max power TS checks : simplify TsActions::resizeWhenOneTS

* Refactor max power TS checks : remove a useless argument to free function invalidateArea

* Refactor max power TS checks : changing invalidateArea's signature and content

In this context, pointer area (as argument) is necessarily not null, because it is used way before we come to this point.
So :
- We make area a reference
- We stop testing the area address (we known it's not null)
- We remove the dead code (when area address is null)

* Refactor max power TS checks : several things (see commit message)

- Move class NumberComparison to separate source files
- Remove return type to function postProcessMaxPowerTS : no more boolean returned, but void.
  Returning boolean is for loading tasks, here we check and possibly change already loaded data.
  We adapt tests, as a consequence
- Removing class TsActions, not useful
-

* Refactor max power TS checks : add forgotten files + renaming

* Refactor max power TS checks : remove case where both TS matrices (pump + gen) have null width, impossible

* Refactor max power TS checks : just removing useless comments

* Refactor max power TS checks : a bot of cleaning

* Refactor max power TS checks : internalize count of generation time series (ror, storage, mingen)

We should not be able to change a TS count without resizing the TS.
This was allowed when because this count was a public data member.
So we make it a private data member, and change it only when resizing the associated data.

* Refactor max power TS checks : renaming

* Refactor max power TS checks : renaming

* Refactor max power TS checks : adapt CI Window workflow to new runner (about python)

* Refactor max power TS checks : define and use of function EqualizeTSsize(...)

* Refactor max power TS checks : rearrange DataSeriesHydro::loadFromFolder, but no major change

* Refactor max power TS checks : remove the case where TS (ror, storage, mingen) have a zero size. Impossible case.

* Refactor max power TS checks : fix bugs on function EqualizeTSsize

* Refactor max power TS checks :  use maxPowerTScount() whenever possible

* Refactor max power TS checks : removing the setter for max power TS number

* Refactor max power TS checks : fix a bug due to removing the setter for max power TS number

* Refactor max power TS checks : gather loading and equalizing hydro mingen TS

* Refactor max power TS checks : split load of ROR, STORAGE and MINGEN

* Refactor max power TS checks : loading Max Power TS : using a generic loading function

* Refactor max power TS checks : fix a bug on loading inflows (height ont correct)

* Refactor max power TS checks : correction after (small) merge

* Refactor max power TS checks : fixing bug on loading inflows was wrong. Fixing it for real now.

* Refactor max power TS checks : remove useless checkMinGenTsNumber

* Refactor max power TS checks : cleaning on the very ugly setHydroModulability (too many args, body too big)

* Refactor max power TS checks : correct derated mode for hydro

* Refactor max power TS checks : some more corrections due to review

* Refactor max power TS checks : some more corrections after review

* Refactor max power TS checks : fixing errors introduced by earlier corrections

* Refactor max power TS checks : more corrections after review

* Refactor max power TS checks : more corrections due to tight review :-)

* Fixing sonar bug

* Code smells

* Use init-statement inside the if statement for hydroSeries ptr

* More code smells

---------

Co-authored-by: NikolaIlic <nikola.ilic@redstork-solutions.com>

nikolaredstork committed:
Bugfix

nikolaredstork committed:
Code smells
